devops/templates/test-run-step-template.yml

```diff
@@ -70,6 +70,14 @@ steps:
   displayName: "Run flake8"
 
 # =================================================
+- ${{ if eq(parameters.testRunType, 'Unit-Deprecation')}}:
+  - script: python -m pytest test/ -Werror::DeprecationWarning -Werror::FutureWarning --ignore=test/install -m "not notebooks" --junitxml=./TEST--TEST.xml -o junit_suite_name="$(Agent.JobName)"
```

Comment:
I actually meant just adding the `-Werror` arguments to the `Unit` line below. Sorry for the misunderstanding.

Reply:
Sure, didn't wanna have it always error. Thought it'd be nice to have the normal tests and just have only one CI run detecting these, for us to see whether they fail because of the warnings or other issues. But could add them here.

Reply:
If your changes above make us 'clean' then I don't see the point in burning the CPU time on a separate pipeline run which effectively just checks for warnings.

If we've a way to go to squelch everything, then maybe a separate pipeline is useful for isolating the issues.